x/wasm/keeper/addresses_test.go

```diff
@@ -3,20 +3,82 @@ package keeper
 import (
 	"encoding/json"
 	"fmt"
+	tmbytes "github.com/cometbft/cometbft/libs/bytes"
 	"testing"
 
-	tmbytes "github.com/cometbft/cometbft/libs/bytes"
 	"github.com/stretchr/testify/require"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-func TestBuildContractAddress(t *testing.T) {
+func TestBuildContractAddressClassic(t *testing.T) {
+	// preserve current Bech32 settings and restore them after test completion
 	x, y := sdk.GetConfig().GetBech32AccountAddrPrefix(), sdk.GetConfig().GetBech32AccountPubPrefix()
+	c := sdk.IsAddrCacheEnabled()
 	t.Cleanup(func() {
 		sdk.GetConfig().SetBech32PrefixForAccount(x, y)
+		sdk.SetAddrCacheEnabled(c)
 	})
+
+	// set custom Bech32 settings
 	sdk.GetConfig().SetBech32PrefixForAccount("purple", "purple")
+	// disable address cache
+	// AccAddress -> String conversion is then slower, but does not lead to errors like this:
+	//   runtime error: invalid memory address or nil pointer dereference
+	sdk.SetAddrCacheEnabled(false)
```

Comment:
you could extract this piece of code and have it into a separate private function e.g. `setup(t *testing.T)` since it is same for both tests.

Reply:
Done.